packages/next/src/server/lib/patch-fetch.ts

```diff
@@ -21,278 +21,273 @@ export function patchFetch({
 
   // @ts-expect-error - we're patching fetch
   // eslint-disable-next-line no-native-reassign
-  fetch = getTracer().wrap(
-    AppRenderSpan.fetch,
-    {
-      kind: SpanKind.CLIENT,
-    },
-    async (input: RequestInfo | URL, init: RequestInit | undefined) => {
-      const staticGenerationStore = staticGenerationAsyncStorage.getStore()
-      const isRequestInput =
-        input &&
-        typeof input === 'object' &&
-        typeof (input as Request).method === 'string'
-
-      const getRequestMeta = (field: string) => {
-        let value = isRequestInput ? (input as any)[field] : null
-        return value || (init as any)?.[field]
-      }
+  fetch = async (input: RequestInfo | URL, init: RequestInit | undefined) => {
+    let url
+    try {
+      url = new URL(input instanceof Request ? input.url : input)
```

Comment:
where is that logic from?

Reply:
That's pretty ad-hoc. I just needed to get the URL to show it in the span root.

Reply:
`Request` is part of the `RequestInfo` type.